File: week15/week15-3.py
#1208. Get Equal Substrings Within Budget
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        N=len(s) #字串長度
        ans=0
        j=0
        for i in range(N):
            maxCost-=abs(ord(s[i]) - ord(t[i]))
            while maxCost<0:
                maxCost+=abs(ord(s[j])-ord(t[j]))
                j+=1
            ans=max(ans,i-j+1)
            #print(s[i],t[i]);#印出字串陣列

        return ans

File: week15/week15-3.c
///1208. Get Equal Substrings Within Budget
int equalSubstring(char* s, char* t, int maxCost) {
    int N=strlen(s);
    int ans=0,j=0;
    for(int i=0;i<N;i++){
        maxCost-=abs(s[i]-t[i]);
        while(maxCost<0){
            maxCost+=abs(s[j]-t[j]);
            j++;
        }
        ///printf("%c %c\n",s[i],t[i]);
        if(i-j+1>ans) ans=i-j+1;
    }
    return ans;
}